In [2]:
import pandas as pd

In [3]:
file = './dataset/train_data.csv'
df = pd.read_csv(file)

In [12]:
df.iloc[:10, :]

customer_ID         S_2       P_2  \
0  0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...  2017-03-09  0.938469   
1  0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...  2017-04-07  0.936665   
2  0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...  2017-05-28  0.954180   
3  0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...  2017-06-13  0.960384   
4  0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...  2017-07-16  0.947248   
5  0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...  2017-08-04  0.945964   
6  0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...  2017-09-18  0.940705   
7  0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...  2017-10-08  0.914767   
8  0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...  2017-11-20  0.950845   
9  0000099d6bd597052cdcda90ffabf56573fe9d7c79be5f...  2017-12-04  0.868580   

       D_39       B_1       B_2       R_1       S_3      D_41       B_3  ...  \
0  0.001733  0.008724  1.006838  0.009228  0.124035  0.008771  0.004709  ...   
1  0.005775  0.004923  1.000653  0.006151  0.126750  0.000798  0.002714  ...   
2  0.091505  0.021655  1.009672  0.006815  0.123977  0.007598  0.009423  ...   
3  0.002455  0.013683  1.002700  0.001373  0.117169  0.000685  0.005531  ...   
4  0.002483  0.015193  1.000727  0.007605  0.117325  0.004653  0.009312  ...   
5  0.001746  0.007863  1.005006  0.004220  0.110946  0.009857  0.009866  ...   
6  0.002183  0.018859  1.008024  0.004509  0.103329  0.006603  0.000783  ...   
7  0.003029  0.014324  1.000242  0.000263  0.108115  0.009527  0.007836  ...   
8  0.009896  0.016888  1.003995  0.001789  0.102792  0.002519  0.009817  ...   
9  0.001082  0.001930  1.007504  0.001772  0.100470  0.004626  0.006073  ...   

   D_136  D_137  D_138     D_139     D_140     D_141  D_142     D_143  \
0    NaN    NaN    NaN  0.002427  0.003706  0.003818    NaN  0.000569   
1    NaN    NaN    NaN  0.003954  0.003167  0.005032    NaN  0.009576   
2    NaN    NaN    NaN  0.003269  0.007329  0.000427    NaN  0.003429   
3    NaN    NaN    NaN  0.006117  0.004516  0.003200    NaN  0.008419   
4    NaN    NaN    NaN  0.003671  0.004946  0.008889    NaN  0.001670   
5    NaN    NaN    NaN  0.001924  0.008598  0.004529    NaN  0.000674   
6    NaN    NaN    NaN  0.001336  0.004361  0.009387    NaN  0.007727   
7    NaN    NaN    NaN  0.002397  0.008452  0.005553    NaN  0.001831   
8    NaN    NaN    NaN  0.009742  0.003968  0.007945    NaN  0.008722   
9    NaN    NaN    NaN  0.003611  0.009607  0.007266    NaN  0.008763   

      D_144     D_145  
0  0.000610  0.002674  
1  0.005492  0.009217  
2  0.006986  0.002603  
3  0.006527  0.009600  
4  0.008126  0.009827  
5  0.002223  0.002884  
6  0.007661  0.002225  
7  0.009616  0.007385  
8  0.004369  0.000995  
9  0.004753  0.009068  

[10 rows x 190 columns]

### Project Proposal Meeting
1. Data Understanding
    - there are 190 columns => 188 features (since the first column is index and the 2nd column is customer ID) 
    - 5,531,451 data points
    - 1 feature is the date. Could we elimiate it?
    - some feature has "NaN" value. How do we represent it? 
        run regression to predict it (statiscal analysis: modal, meadian, put the data on a distribution and see where the missing data would lie)
        0
        remove the columns
    
    => find a paper to read about how to deal with missing values

2. Papers:
    https://www.kaggle.com/competitions/amex-default-prediction/discussion/327135   

3. Team meeting for proposal recording on Thursday 10/6 at 5:30 PM

4. Trello Board: https://trello.com/b/F0tEjnUX/ml-project

5. Gantt Chart: https://docs.google.com/spreadsheets/d/1NwSPawBI_k9x3xHloXmnbROMbCaqwuFalB0XVgNrCJ8/edit?usp=sharing